quinn-proto/src/connection/stats.rs

```diff
@@ -0,0 +1,80 @@
+/// Statistics about packets transmitted on a connection
+#[derive(Default)]
+pub struct PacketTransmissionStats {
+    /// The amount of packets transmitted on a connection
+    pub packets: u64,
+    /// The total amount of bytes transmitted on a connection
+    pub bytes: u64,
+}
+
+impl PacketTransmissionStats {
+    /// Calculates the average packet size
+    pub fn avg_packet_size(&self) -> u64 {
```

Comment:
Does this really need to be a helper method?

Reply:
What else do you have in mind? A field that gets updated with every packet?

Reply:
Just leaving it for callers to compute if they want it.

Reply:
I’m not in favor of that. I found it immensely helpful to just debug print this struct and get all the information I need without adding any more external code. 

And packet utilization can for example be a more interesting metric than pure overall bytes or number of packets. 

I think there will be more metrics which will fall into the same category. Eg average bps, average packet loss, etc. I wouldn’t want all consumers to have to recalculate those on their end.

Reply:
How do we draw the line between derived statistics that are worth providing out of the box and ones that should be left to downstream?

Reply:
That can be discussed as part of each CR which wants to add more, right?
I think the main criteria should be: Is this statistic understandable for all users?
I don't think having more is any issue, since derived statistics would always be zero cost, apart from when printing them (which you probably won't do unless you are debugging).

Reply:
My concern is that the set of derived statistics that are plausibly useful to someone is very large, and I don't want to end up with hundreds/thousands of lines of statistical analysis code that we have to maintain in the long run. The cost isn't computational, but in maintenance effort (and to a lesser extent, compile time). That sort of logic really belongs in metrics pipelines that the source data can easily be dumped into, and where dependencies for data structures and algorithms to support analysis are justifiable.

That doesn't mean we necessarily can't have helpers for some overwhelmingly simple and useful stuff, but "anything comprehensible" is too low a bar.

Reply:
@Matthias247 I think this is the only open question here. I'm with @Ralith on the maintenance aspects. Can you please remove the method?